server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/MockMessageHandler.java

```diff
@@ -62,6 +62,28 @@ public void behave(T input) throws RejectException {
         }
     }
 
+    class SMTPBehaviorRepositoryUpdater<T> implements Behavior<T> {
+
+        private final SMTPBehaviorRepository behaviorRepository;
+        private final MockBehavior<T> actualBehavior;
+
+        SMTPBehaviorRepositoryUpdater(SMTPBehaviorRepository behaviorRepository, MockBehavior<T> actualBehavior) {
+            this.behaviorRepository = behaviorRepository;
+            this.actualBehavior = actualBehavior;
+        }
+
+        @Override
+        public void behave(T input) throws RejectException {
+            try {
+                actualBehavior.behave(input);
+            } catch (RejectException e) {
+                behaviorRepository.decreaseRemainingAnswers(actualBehavior.behavior);
```

Comment:
Why do we do that only when we reject?

Does this means "answerCount" is not working when not rjecting? When 250? This don't seems right.

(don't you want a "final" block here?)